### PR TITLE
Server configuration

### PR DESCRIPTION
**Pull request by DavidBronfen**
#43

### CODE REVIEW

**Comment by DavidBronfen** [comment]
@nirgn975 The tests here are green but it doesn't test the server.
Ready to review.

File: client/src/app/modules/recipes/recipes.module.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { RecipesEffects } from './effects/recipes';
     StoreDevtoolsModule.instrumentOnlyWithExtension(),
     EffectsModule.run(RecipesEffects),
     StarRatingModule.forRoot(),
+
```
Comment (nirgn975) [inline client/src/app/modules/recipes/recipes.module.ts]: Why?

File: server/config/production.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  // Disable logging for production.
+  logging: false
```
Comment (nirgn975) [inline server/config/production.js]: Also, add `seed: false`, you don't want to seed the production db, and __explicit is better than implicit__

Comment (nirgn975) [reply server/config/production.js]: Also add
```
db: {
  url: 'mongodb//localhost/retrogusto-prod',
}
```

File: server/config/testing.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  // Disable logging for testing.
+  logging: false,
+  db: {
+    url: 'mongodb://localhost/retrogusto-test'
```
Comment (nirgn975) [inline server/config/testing.js]: Add a comma in the end ([ES Lint comma-dangle](https://eslint.org/docs/rules/comma-dangle))

File: server/config/testing.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  }
```
Comment (nirgn975) [inline server/config/testing.js]: Add a comma in the end ([ES Lint comma-dangle](https://eslint.org/docs/rules/comma-dangle))

File: server/config/development.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  // Enable logging for development.
+  logging: true,
+  seed: false, // Change to true when seed will be set.
+  db: {
+    url: 'mongodb://localhost/retrogusto'
```
Comment (nirgn975) [inline server/config/development.js]: Add a comma in the end ([ES Lint comma-dangle](https://eslint.org/docs/rules/comma-dangle))

File: server/config/development.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  }
```
Comment (nirgn975) [inline server/config/development.js]: Add a comma in the end ([ES Lint comma-dangle](https://eslint.org/docs/rules/comma-dangle))

File: server/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Retrogustoe",
+  "version": "1.0.0",
+  "description": "Retrogusto server",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "nodemon index.js"
+  },
+  "author": "",
```
Comment (nirgn975) [inline server/package.json]: Add your name to it, or remove this line.

File: server/server.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+const logger = require('./util/logger');
+
+// db.url is different depending on NODE_ENV
+mongoose.connect(config.db.url);
+
+if (config.seed) {
+  require('./util/seed');
+}
+
+require('./middleware/app.middleware');
```
Comment (nirgn975) [inline server/server.js]: Should be `require('./middleware/app.middleware')(app);`, your `export` line in `app.middleware` expect to receive an `app`.

File: server/util/logger.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    const tag = '[ ✨ LOG ✨ ]'.green;
+    // arguments is an array like object with all the passed
+    // in arguments to this function
+    const args = _.toArray(arguments)
+      .map((arg) => {
+        if(typeof arg === 'object') {
+          // turn the object to a string so we
+          // can log all the properties and color it
+          const string = JSON.stringify(arg, null, 2);
+          return tag + '  ' + string.cyan;
```
Comment (nirgn975) [inline server/util/logger.js]: Use ES2015, should be 
```
return `${tag} ${string.cyan}`;
```

File: server/util/logger.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        } else {
+          return tag + '  ' + arg.cyan;
```
Comment (nirgn975) [inline server/util/logger.js]: Use ES2015, should be 
```
return `${tag} ${arg.cyan}`;
```

File: server/util/logger.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    // and the new colored args :)
+    consoleLog.apply(console, args);
+  },
+
+  error: () => {
+    const args = _.toArray(arguments)
+      .map((arg) => {
+        arg = arg.stack || arg;
+        const name = arg.name || '[ ❌ ERROR ❌ ]';
+        const log = name.yellow + '  ' + arg.red;
```
Comment (nirgn975) [inline server/util/logger.js]: Use ES2015, should be 
```
cont log = `${name.yellow} ${arg.red}`;
```

File: client/src/app/modules/recipes/recipes.module.ts (hunk 1 above)
Comment (DavidBronfen) [reply client/src/app/modules/recipes/recipes.module.ts]: Because I can! 💪

**Pull request merged**